chapter2/tests/t19/terraform/_modules/container/main.tf

```diff
@@ -0,0 +1,25 @@
+resource "docker_image" "cont" {
+  name         = var.container_image
+  keep_locally = var.container_image_keep_locally
+}
+
+resource "docker_container" "cont" {
+  image = docker_image.cont.image_id
+  name  = var.container_name
+  ports {
+    internal = var.container_ports.internal
+    external = var.container_ports.external
+  }
+  dynamic "volumes" {
+    for_each = var.container_volume_hostpath != "" && var.container_volume_path != "" ? [1] : []
```

Comment:
try to use `count` to get rid of `[1]:[]`,
so 
```
count = condition ? 1 : 0
```

as you're not `each.XXX` functionality at all.

Reply:
instead it would be good to have a list var
`var.container_volumes = list(map)` that you pass into module, and then the use of foreach is a  good solution as you can parameterize that dynamic block (i.e. have multiple volumes) by external var.

Reply:
> try to use `count` to get rid of `[1]:[]`, so
> 
> ```
> count = condition ? 1 : 0
> ```
> 
> as you're not `each.XXX` functionality at all.

@dmitry-mightydevops 
Done

Reply:
> instead it would be good to have a list var `var.container_volumes = list(map)` that you pass into module, and then the use of foreach is a good solution as you can parameterize that dynamic block (i.e. have multiple volumes) by external var.


@dmitry-mightydevops 
Done